src/Storages/Kafka/KafkaConsumer.cpp

```diff
@@ -462,6 +468,11 @@ ReadBufferPtr KafkaConsumer::consume()
         }
         else
         {
+            if (is_streaming
+                && new_messages.size() == 1
+                && new_messages[0].get_error() == RD_KAFKA_RESP_ERR_UNKNOWN_TOPIC_OR_PART)
```

Comment:
1. From our view, there are lots of error handling problems / challenges in the existing CH Kafka table engine. This is one reason that we didn't use it in T+ Proton. We rewrite completely based on our past experience.
2. We try hard to to keep backward compatibility with existing Kafka table engine, but still tries to handle more errors gracefully like topics got deleted. However, we would like to minimize the scope of this enhancement since that deserves a different PR.